src/log/access_log.h

```diff
@@ -13,7 +13,9 @@
 
 #include "LogTags.h"
 
-void fvdbCountVia(const char *key);
+class String;
+void fvdbCountVia(const String &key);
```

Comment:
> I would actually propose to do something different, which is to unconditionally enable this code, for the purpose of simplifying our spaghetti.

My simple suggestion unconditionally enables this code _in the caller_ so it is a big step towards your preferred solution AFAICT.

Unconditionally enabling this whole feature is not a good idea right now because it is too expensive performance- and memory-wise -- there are simply not enough controls to prevent trivial abuse by outsiders AFAICT, especially in a reverse proxy setting. Replacing `#ifdef` with a squid.conf setting would be a good idea though (and my simple suggestion would help with that too by moving the decision away from the caller).

I agree that any non-trivial changes can be viewed as being outside this PR scope, especially changes that alter Squid build- or runtime configuration.